### PR TITLE
update checkpoint docs

### LINKED ISSUE #775: Checkpoint naming broken
## 🐛 Bug

I would like to be able to save checkpoints with custom names that include the value of my `val_loss`, ie. `path/epoch_2-val_loss_0.2.hdf5 `. The [documentation](https://pytorch-lightning.readthedocs.io/en/latest/pytorch_lightning.callbacks.html#pytorch_lightning.callbacks.ModelCheckpoint) for `ModelCheckpoint` suggests that this is possible using the `filepath` argument. This does not appear to be the case, since the [source code](https://github.com/PyTorchLightning/PyTorch-Lightning/blob/master/pytorch_lightning/callbacks/pt_callbacks.py#L247) calls `os.mkdirs(filepath)`. I have also tried using the `prefix` argument, but it doesn't seem to be possible to pass it a format string containing a variable. 

## Expected behavior

The documentation claims that `filepath='{epoch:02d}-{val_loss:.2f}.hdf5'` will save a checkpoint at `/path/epoch_2-val_loss_0.2.hdf5`. Instead, it saves a checkpoint at `{epoch:02d}-{val_loss:.2f}.hdf5/_ckpt_epoch_1.ckpt`. 

The issues in the documentation are two-fold: 
-- It suggests that `filepath` can contain the directory + name of the checkpoint, when it seems like it should only contain the directory specifying where to save. 
-- It suggests that it can 'contain named formatting options to be auto-filled', which also doesn't seem to be the case. 

Is it possible to achieve this functionality with the `prefix` argument instead? If so, how?

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?
Fixes #775. Revisit checkpoint documentation...

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/model_checkpoint.py [hunk 1]
```diff
@@ -127,35 +136,32 @@ def on_validation_end(self, trainer, pl_module):
...
+                            log.info('Epoch %05d: %s was not in top %i', epoch, self.monitor, self.save_top_k)
 
             else:
                 if self.verbose > 0:
-                    log.info(f'\nEpoch {epoch:05d}: saving model to {filepath}')
+                    log.info('Epoch %05d: saving model to %s', epoch, filepath)
                 self._save_model(filepath)
 
-    def _do_check_save(self, filepath, current, epoch):
+    def _do_check_save(self, filepath: str, current, epoch):
```
Comment (jeremyjordan) [inline pytorch_lightning/callbacks/model_checkpoint.py]: i like the addition of type hints, can we do it for all of the args though? (ie. `current` and `epoch`)

**Comment by Borda** [comment]
shall we make it rather `model-name_epoch=02_val_loss=0.36.ckpt` and if there is already something with `model-name` then it creates `model-name-v1_epoch=02_val_loss=0.36.ckpt` that would make sense more... but we can move it to next release and just keep this one now @williamFalcon

**Comment by williamFalcon** [comment]
```
________________________________________________________________________________ test_resume_from_checkpoint_epoch_restored ________________________________________________________________________________

tmpdir = local('/tmp/pytest-of-waf251/pytest-48/test_resume_from_checkpoint_ep0')

    def test_resume_from_checkpoint_epoch_restored(tmpdir):
        """Verify resuming from checkpoint runs the right number of epochs"""
        import types

        tutils.reset_seed()

        hparams = tutils.get_hparams()

        def _new_model():
            # Create a model that tracks epochs and batches seen
            model = LightningTestModel(hparams)
            model.num_epochs_seen = 0
            model.num_batches_seen = 0

            def increment_epoch(self):
                self.num_epochs_seen += 1

            def increment_batch(self, _):
                self.num_batches_seen += 1

            # Bind the increment_epoch function on_epoch_end so that the
            # model keeps track of the number of epochs it has seen.
            model.on_epoch_end = types.MethodType(increment_epoch, model)
            model.on_batch_start = types.MethodType(increment_batch, model)
            return model

        model = _new_model()

        trainer_options = dict(
            show_progress_bar=False,
            max_epochs=2,
            train_percent_check=0.65,
            val_percent_check=1,
            checkpoint_callback=ModelCheckpoint(tmpdir, save_top_k=-1),
            logger=False,
            default_save_path=tmpdir,
            early_stop_callback=False,
            val_check_interval=0.5,
        )

        # fit model
        trainer = Trainer(**trainer_options)
        trainer.fit(model)

        training_batches = trainer.num_training_batches

        assert model.num_epochs_seen == 2
        assert model.num_batches_seen == training_batches * 2

        # Other checkpoints can be uncommented if/when resuming mid-epoch is supported
        checkpoints = sorted(glob.glob(os.path.join(trainer.checkpoint_callback.dirpath, '*.ckpt')))

        for check in checkpoints[::2]:
            next_model = _new_model()
            state = torch.load(check)

            # Resume training
            trainer_options['max_epochs'] = 4
            new_trainer = Trainer(**trainer_options, resume_from_checkpoint=check)
            new_trainer.fit(next_model)
>           assert state['global_step'] + next_model.num_batches_seen == training_batches * 4
E           assert 140 == 160
E             -140
E             +160
```

**Comment by Borda** [comment]
it is hard to replicate it locally, it seems the names on server are different...

**Comment by Borda** [comment]
it shall be fixed now, @williamFalcon ^^

**Pull request merged**